src/Slider.tsx

```diff
@@ -1,25 +1,31 @@
 import classNames from 'classnames';
-import type React from 'react';
-import type { MouseEvent as ReactMouseEvent, PropsWithChildren } from 'react';
-import { useRef, useEffect, Children, useCallback, useState } from 'react';
+import type { MouseEvent as ReactMouseEvent } from 'react';
+import { Children, forwardRef, PropsWithChildren, useCallback, useEffect, useImperativeHandle, useRef, useState } from 'react';
 import { NextButton } from './Components/Controls/NextButton';
 import { PreviousButton } from './Components/Controls/PreviousButton';
 import { NavigationDirection, useSlider, Visibility } from './Hooks/UseSlider';
 import './Slider.scss';
 
+export namespace SliderTypes {
+    export interface Settings {
+        // Sets whether the navigation buttons (next/prev) are no longer rendered
+        hideNavigationButtons?: boolean;
+        initialSlideIndex?: number;
+    }
+
+    export interface API {
+        scrollToSlide: (index: number, smooth: boolean) => void;
```

Comment:
`smooth: boolean` seems like a weird property to have, that's not very extendible and will probably introduce a BC break at some point.

I would prefer something like `scrollBehaviour: 'auto' | 'smooth';` (or create an actual enum).